cws-service/src/main/java/jpl/cws/console/WorkerMonitorBackgroundThread.java

```diff
@@ -102,17 +106,37 @@ public void run() {
 					//
 					externalWorkersThatWentDown = schedulerDbService.detectDeadExternalWorkers(THRESHOLD_MILLIS_FOR_DEAD_WORKER);
 				}
-				
+
+				// Successful thread iteration, so reset failure variables back to nominal
+				failures = 0;
+				failWait = 100;
 				
 			} catch (InterruptedException e) {
 				log.warn("Interrupted. Must be shutting down..");
-				break;
+				break; // break out of loop to stop this thread
 			}
 			catch (Throwable e) {
-				log.error("Unexpected error occured.  Details: ", e);
-				break;
+				failures++;
+				if (failures > maxFailures) {
+					log.error("Unexpected error occurred. Exiting WorkerMonitorBackgroundThread, as there have already been " +
+							failures + " failures. Details: ", e);
+					break; // break out of loop to abort this thread
+				}
+				else {
+
+					// wait, before trying again
+					try {
+						log.warn("Unexpected error occurred.  Waiting " + failWait + " milliseconds, before trying next attempt (" +
+								failures + "/" + maxFailures+") ...", e);
+						Thread.sleep(failWait);
+					} catch (InterruptedException ex) {
+						ex.printStackTrace();
+					}
+
+					failWait = failWait * 2; // update exponential backoff wait time
```

Comment:
This exponential backoff comes out to `(100×2^20)/(3.6×10^6) = 29.1hrs` in the worst case.. so just over a day of allowable database downtime. Seems ok!